### PR TITLE
Fix css embed in JSX when using (only) a spread attribute

### PR DESCRIPTION
**Pull request by eelco**
## Description

Prettier *crashes* on this JSX/TSX code:
```
<style {...spread}>{`.{}`}</style>
```

This is fixed by improving the detection mechanism used in `isStyledJsx`.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/embed/css.js [hunk 1]
```diff
@@ -75,7 +75,10 @@ function isStyledJsx({ node, parent, grandparent }) {
       grandparent.type === "JSXElement" &&
       grandparent.openingElement.name.name === "style" &&
       grandparent.openingElement.attributes.some(
-        (attribute) => attribute.name.name === "jsx",
+        (attribute) =>
+          (attribute.type === "JSXAttribute" &&
+            attribute.name.name === "jsx") ||
+          attribute.type === "JSXSpreadAttribute",
```
Comment (fisker) [inline src/language-js/embed/css.js:81]: I don't think we should add this `JSXSpreadAttribute` here.

Comment (fisker) [inline src/language-js/embed/css.js:81]:
```suggestion
      (attribute) =>
        attribute.type === "JSXAttribute" && attribute.name.name === "jsx",
````

Comment (eelco) [reply src/language-js/embed/css.js:81]: Why not? Without it the CSS within the template string will not format in the example given in the description and the test.

Comment (eelco) [reply src/language-js/embed/css.js:81]: That said, I’m not entirely sure why the `isStyledJsx` function looks at the attributes of the `style` element. Perhaps that whole check can be removed?

Comment (fisker) [reply src/language-js/embed/css.js:81]: `jsx` is required in [`styled-jsx`](https://github.com/vercel/styled-jsx). We don't want to break something we don't know.

Comment (eelco) [reply src/language-js/embed/css.js]: Updated the code and the snapshot.

**Review by fisker — APPROVED** [review]
Thank you!

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**